projects/ngx-d3timeline/src/lib/scale-types.ts

```diff
@@ -1,5 +1,6 @@
 import { ScaleBand, ScaleTime } from 'd3-scale';
 
 export type BandScale = ScaleBand<string>;
+export type InverseBandScale = (value: number) => string;
```

Comment:
Not convinced about this name, it sounds like it’s a Bandscale like  object, but it’s actually a function. And the name doesn’t really say what the function does.

I think it’s taking a range/pixel value in the band scale and covering it to a domain value?

It seems like it’s closely aligned with the actual Bandscale object, maybe we have that and add this to it somehow? Not sure...

Otherwise we name it something different, can think of anything great atm, something like bandScaleRangeToDomain?

Reply:
Is it because you can use Bandscale like

Domain = bandScale(range);  // not sure this is how it works

Reply:
Exactly, a scale is just a function that maps domain to range. BandScale maps from a domain that has type band (d3 terminology) i.e. the distinct resources in the data.

so a scale is just f(domain) => range

The inverse of a function is function that reverses another function. So the inverse of

f(domain) => range 

is by definition

f(range) => domain

Re classes, functions are objects in Javascript. In fact our BandScale and TimeScale types are also functions.

Reply:
Got it